version: '3'
services:
  reverse-proxy:
          image: traefik
          command:
              - "--providers.docker.endpoint=unix:///var/run/docker.sock"  
              - "--providers.docker=true"
              - "--api.insecure=true"
              - "--providers.docker.exposedbydefault=false"
              - "--providers.docker.network=front"
              - "--entrypoints.web.address=:80"
          ports:
                # The HTTP port
                - "80:80"
                # The Web UI (enabled by --api.insecure=true)
                - "8080:8080"
          volumes:
          # So that Traefik can listen to the Docker events
          - /var/run/docker.sock:/var/run/docker.sock
          networks:
              - front
  database:
    image: postgres:alpine
    env_file:
      - .env/postgres.env
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
    healthcheck:
            test: [ "CMD", "pg_isready", "-q", "-d", "${DB_NAME}", "-U", "postgres" ]
            timeout: 45s
            interval: 10s
            retries: 10
    networks:
            - back
  django:
    image: smvsite/smvsite
    env_file:
      - .env/django.env
    depends_on:
            database:
                    condition:  service_healthy
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.django.rule=Host(`127.0.0.1`)"
      - "traefik.http.services.django.loadbalancer.server.port=8000"
      - "traefik.http.routers.traefik.entrypoints=web"
      - "traefik.port=8000"
    networks:
            - back
            - front
networks:
        back:
             driver: bridge
             ipam:
                config:
                  - subnet: 10.5.0.0/16
                    gateway: 10.5.0.1
        front:
            driver: bridge
            ipam:
               config:
                 - subnet: 10.6.0.0/16
                   gateway: 10.6.0.1
